lib/foreman/foreman_url_renderer.rb

```diff
@@ -9,7 +9,12 @@ def default_url_options
     end
 
     # returns the URL for Foreman based on the required action
-    def foreman_url(action = 'provision', params = {})
+    def foreman_url(action, params = {})
+      if action.nil?
```

Comment:
If `action` parameter isn't passed, `action` won't be `nil` but rather an error will be thrown due to mismatching number of arguments. you can change the default for `action` to `nil` for this to work.

Reply:
Damn yeah. Let's do this and then we can actually change this after this is deprecated. Rebased.